### PR TITLE
Enrich Pull-A-Part vehicle ingestion with extended details and images

### PR DESCRIPTION
**Pull request by AugusDogus**
## Summary
- Enriches Pull-A-Part inventory ingestion with per-vehicle extended info and image lookups.
- Updates canonical transform logic to derive engine, transmission, trim, color, image URL, and normalized available dates from richer source data.
- Adds concurrency-limited enrichment in the connector and skips duplicate VINs before fetching details.
- Adds unit coverage for enrichment-driven fields and date fallback behavior.

## Testing
- `bun test src/server/ingestion/pullapart-transform.test.ts`
- Not run: full `bun run check`
- Not run: integration verification against live Pull-A-Part endpoints

## Summary by CodeRabbit

* **New Features**
  * Enriched vehicle ingestion now fetches extended vehicle details and images; missing or "no data" responses are treated as null.

* **Refactor**
  * Improved deduplication and concurrent enrichment with controlled concurrency, safer fallback logic, and standardized parsing/normalization for color, trim, engine/transmission, and available dates.
  * Enrichment failures are logged and mapped to null; batches only include successfully enriched vehicles.

* **Tests**
  * Added tests covering enrichment, fallbacks, precedence, error reporting, and date edge cases.

### CODE REVIEW

**Comment by AugusDogus** [comment]
@coderabbitai review

**Comment by AugusDogus** [comment]
@coderabbitai review

File: src/server/ingestion/pullapart-connector.ts [hunk 1]
```diff
@@ -149,14 +153,58 @@ export function streamPullapartInventory<E, R>(options: {
...
+                    );
+                    return null;
+                  }),
+                ),
+              );
+
+              return transformPullapartVehicle(row, location, geo, {
+                detail,
+                imageUrl,
+              });
```
[the remark this answers is not in the document]
Comment (AugusDogus) [reply src/server/ingestion/pullapart-connector.ts:181]: @coderabbitai was this issue resolved?

[the remark this answers is not in the document]
Comment (AugusDogus) [reply src/server/ingestion/pullapart-connector.ts:181]: @cursoragent can you fix?
